src/App.tsx

```diff
@@ -1,19 +1,67 @@
-import { useState, useEffect, useRef } from 'react'
+import { useState, useEffect, useRef, useCallback } from 'react'
+import { DndProvider, useDrag, useDrop } from 'react-dnd'
+import { HTML5Backend } from 'react-dnd-html5-backend'
 import './App.css'
 
 interface Todo {
-  id: number;
+  id: string;
   text: string;
   completed: boolean;
   dueDate: Date;
 }
 
+interface DraggableItemProps {
+  id: string;
+  index: number;
+  moveItem: (dragIndex: number, hoverIndex: number) => void;
+  children: React.ReactNode;
+  group: 'today' | 'tomorrow' | 'upcoming';
+}
+
+const DraggableItem: React.FC<DraggableItemProps> = ({ id, index, moveItem, children, group }) => {
```

Comment:
This drag and drop functionality isn't working properly. I can pick up an item but when I hover over another item while dragging, the other items don't re-order.